### ISSUE TITLE
No setup information found for ghc-8.4.3 on your platform (`linux64-gmp4`)

### ISSUE THREAD

**Issue by ababkin** [commercialhaskell/stack#4161]
### General summary/comments (optional)

Currently getting this in my CentOS 6.9 based build:
```
Downloading lts-12.0 build plan ...
Downloaded lts-12.0 build plan.
No setup information found for ghc-8.4.3 on your platform.
This probably means a GHC bindist has not yet been added for OS key 'linux64-gmp4'.
Supported versions: ghc-7.8.4, ghc-7.10.2, ghc-7.10.3, ghc-8.0.1, ghc-8.0.2, ghc-8.2.1, ghc-8.2.2, ghc-8.4.2
```

### Steps to reproduce

For example:

`Dockerfile`:
```
FROM       centos:6.9
## Install dependencies
RUN yum install -y curl\
      gcc \
      gmp-devel \
      pcre-devel \
      perl \
      tar \
      which \
      xz \
      zlib-devel \
      bzip2-devel \
    && yum clean all --releasever=6 \
    && ln -s /lib64/libtinfo.so.5 /lib64/libtinfo.so

ENV GHC_VERSION=8.4.3

## Install stack
ARG STACK_VER=1.7.1
RUN mkdir -p /opt/stack/bin
RUN curl -L https://github.com/commercialhaskell/stack/releases/download/v$STACK_VER/stack-$STACK_VER-linux-x86_64-gmp4.tar.gz | tar xz --wildcards --strip-components=1 -C /opt/stack/bin '*/stack'
ENV PATH /opt/stack/bin/:$PATH
RUN stack setup $GHC_VERSION
```
run:
`docker build .`

### Expected

docker build to succeed and provide the GHC 8.4.3 inside the container

### Actual

see the error above

### Stack version

`https://github.com/commercialhaskell/stack/releases/download/v1.7.1/stack-1.7.1-linux-x86_64-gmp4.tar.gz`

### Method of installation

see the `Dockerfile` above

### Comments

looks like the GHC 8.4.3 bindist is missing for Centos 6
https://github.com/fpco/stackage-content/blob/6b816d9287b408e24c2aa0d19477fab0f5444e8b/stack/stack-setup-2.yaml#L369

**Comment by ababkin** [comment]
attn: @borsboom

**Comment by dbaynard** [comment]
Have you tried installing tinfo or ncurses? There are builds of 8.4.3 corresponding to `linux64-tinfo6` and `linux64-ncurses6`.

**Comment by ababkin** [comment]
Thanks, no I have not. Is there a way to force these OS keys? (I'm under impression that `stack` just decides on `linux64-gmp4` when seeing my CentOS 6.9)

**Comment by dbaynard** [comment]
I'm not familiar with centos but I believe you'll need to add one of [these](https://rpmfind.net/linux/rpm2html/search.php?query=libtinfo.so.6()(64bit)) to your docker dependencies and it should do it automatically. I don't think libtinfo5 will do it, though I've never investigated that part of the code.

**Comment by dbaynard** [comment]
Indeed; thanks @ncaq. @borsboom should there be such a bindist for centos?

**Comment by ababkin** [comment]
meanwhile and in case CentOS won't be supported now or in the future, what would be a way to use another custom built GHC (i.e. not from the set of bindists that are currently supported) with stack?

**Comment by dbaynard** [comment]
The easiest is to install a system ghc, and use `system-ghc: true`. Then you will need to ensure the correct version is available for the resolver you wish to use. The stackage page for the resolver (e.g. <https://www.stackage.org/lts-12.0>) will say at the top which ghc version you need.

**Comment by ababkin** [comment]
I was wondering if I could use an option in `stack` that would allow me to use the system ghc _without_ modifying the `stack.yaml` file. Reason is that I use this `stack.yaml` to build the project on variety of platforms and I don't want to use the system ghc on all of them, only on this CentOS one.
I've tried `stack config set system-ghc --global true` but unfortunately that didn't work and I get the same error when compiling project with `stack`:
```
Downloading lts-12.1 build plan ...
Downloaded lts-12.1 build plan.
No setup information found for ghc-8.4.3 on your platform.
This probably means a GHC bindist has not yet been added for OS key 'linux64-gmp4'.
Supported versions: ghc-7.8.4, ghc-7.10.2, ghc-7.10.3, ghc-8.0.1, ghc-8.0.2, ghc-8.2.1, ghc-8.2.2, ghc-8.4.2
The command '/bin/sh -c cd /tmp/symbiont-node/src/capybara; stack clean' returned a non-zero code: 1
```

**Comment by dbaynard** [comment]
You should change the `~/.stack/config.yaml` file to say [`system-ghc: true`](https://docs.haskellstack.org/en/stable/yaml_configuration/#system-ghc).

**Comment by ababkin** [comment]
I see, thank you
